### DISCUSSION TITLE
useContractRead is not working with Proxy contract

### DISCUSSION THREAD

**Discussion by sanjaybip** [wevm/wagmi#1940] [Q&A] [answered]
I am trying to read vai a Transparent proxy contract deployed on BSC Testnet. 

```js
import { xyzTokenAbi } from "../../constants/xyzABIs"; // This is the proxy contract ABI (taken from estnet.bscscan.com)
import { xyzTokenAddress } from "../../constants/xyzAddresses" // This is the proxy contract address
......
const { data } = useContractRead({
    address: xyzTokenAddress,
    abi: zyzTokenAbi,
    functionName: "myFun", // This function exist in implementation contract
 });
 ```
 So, the ABI and contract address is from proxy contract. But the `functionName` is from implementation contract.
 
 When I do `console.log(data)` is see `undefined` in my console.

**Comment by greg-schrammel** [comment, marked as answer]
you must use the implementation abi

**Reply by sanjaybip** [reply]
Thanks, I discovered the answer last night with the same advice. I need to still test and implement to confirm. But thanks for confirming and answering

**Comment by denniarems** [comment]
What you meant by implementation abi , can you explain with an example ?
